source/blog/2016-12-06-part1-install-origin-on-f25-atomic-host.html.md

```diff
@@ -0,0 +1,357 @@
+self explanatory but the `openshift_master_default_subdomain` might need
+a little more explanation. Basically the value of this needs to be a
+[Wildcard DNS Record](https://en.wikipedia.org/wiki/Wildcard_DNS_record)
+so that any domain can be prefixed onto the front of the record and it
+will still resolve to the same IP address. We have decided to use a free
+service called [xip.io](http://xip.io/) so that we don't have to set up
+wildcard DNS just for this example.
+
+So for our example, a domain like `app1.54.204.208.138.xip.io` will
+resolve to IP address `54.204.208.138`. A domain like
+`app2.54.204.208.138.xip.io` will also resolve to that same address.
+These requests will come in to node `54.204.208.138`, which is one of
+our worker nodes where a *router* (HAProxy) is running. HAProxy will
+route the traffic based on the domain used (`app1` vs `app2`, etc) to
+the appropriate service within OpenShift.
+
+OK, next up in our inventory file we have some auth settings:
+
+    # enable htpasswd auth
+    openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
+    openshift_master_htpasswd_users={'admin': '$apr1$zgSjCrLt$1KSuj66CggeWSv.D.BXOA1', 'user': '$apr1$.gw8w9i1$ln9bfTRiD6OwuNTG5LvW50'}
+
+You can use a [multitude of authentication
+providers](https://docs.openshift.com/enterprise/3.0/admin_guide/configuring_authentication.html)
+with OpenShift. The above statements say that we want to use `htpasswd`
+for authentication and we want to create two users. The password for the
+`admin` user is `OriginAdmin`, while the password for the `user` user is
+`OriginUser`. We generated these passwords by running `htpasswd` on the
+command line like so:
+
+    $ htpasswd -bc /dev/stdout admin OriginAdmin
+    Adding password for admin user
+    admin:$apr1$zgSjCrLt$1KSuj66CggeWSv.D.BXOA1
+    $ htpasswd -bc /dev/stdout user OriginUser
+    Adding password for user user
+    user:$apr1$.gw8w9i1$ln9bfTRiD6OwuNTG5LvW50
+
+OK, now on to the host groups. First up, our `master` nodes:
+
+    # host group for masters
+    [masters]
+    54.175.0.44 openshift_public_hostname=54.175.0.44 openshift_hostname=10.0.173.101
+
+We have used `54.175.0.44` as the hostname and also set
+`openshift_public_hostname` to this same value so that certificates will
+use that hostname rather than a *detected* hostname. I'm also setting
+the `openshift_hostname=10.0.173.101` because there is a
+[bug](https://github.com/golang/go/issues/17967) where the golang
+resolver can't resolve `*.ec2.internal` addresses. This is also
+documented as an
+[issue](https://github.com/openshift/origin/issues/11962) against
+Origin. You shouldn't have to set `openshift_hostname`, but we will be
+doing that here for all of our nodes.
+
+Next up we have the `etcd` host group. We're simply re-using the master
+node:
```

Comment:
add after node " for a single etcd node.  In a production deployment, we'd have several."